include/xtensor/xchunked_array.hpp

```diff
@@ -0,0 +1,91 @@
+#include <vector>
+#include <array>
+#include "xarray.hpp"
+
+template <class Chunk_type>
+class xchunked_array
+{
+    public:
+
+        using const_reference = typename Chunk_type::const_reference;
+
+        template <class... Args>
+        inline const_reference operator()(Args... args) const
+        {
+            return access_impl(args...);
+        }
+
+        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunks):
```

Comment:
I guess this will eventually replaced with a constructor that accepts a list of `chunk_type` elements?

Reply:
I'm not sure, eventually the chunks won't even be instantiated in `xchunked_array` since they might not even fit in memory.

Reply:
I think this (loading memory or not) should be the reponsibility of the chunk_type, not that of the `chunk_array`. For chunked arrays that fit into memory, we can use `xchunked_array<xarray<double>>` for instance, and for those that do not fit into memory we could use `xchunked_array<xlazy_array<double>>` where `xlazy_array` is a new kind of array that will load / unload memory on demand.

Reply:
But even the list of `xlazy_array<double>` might not fit in memory. Even though they don't hold any (data) memory, the number of objects to allocate can be huge, or even infinite. The way I see things is that `xchunked_array` will only manipulate indexes, and a very limited number of (lazily loaded) chunks in memory.

Reply:
Ok, I didn't consider this extreme case. So in this case, we may want an *additional* constructor taking a list of chunks (when that fits in memory), but it won't replace this one.